### DISCUSSION TITLE
Taichi 2.0

### DISCUSSION THREAD

**Discussion by jarmitage** [taichi-dev/taichi#8542] [Ideas]
> @yuanming-hu: Our current challenge (and opportunity) is in determining the next significant milestones for Taichi 2.0. While the future path is not fully defined, we are excited about the possibilities and welcome your thoughts on potential directions. (One possibility is more support for differentiable graphics, given GenAI, leveraging differentiable graphics, nowadays have increasing impact.) – https://github.com/taichi-dev/taichi/discussions/8506#discussioncomment-9773173

What are your thoughts, ideas, hopes and dreams for Taichi 2.0?

**Comment by jarmitage** [comment]
Targeting WebGPU would be great for many use cases cc @AmesingFlank https://github.com/taichi-dev/taichi/issues/3781

**Comment by bgailleton** [comment]
If this is still active, I'd love to see in 2.0:

- some sort of memory pool (basically being able to deallocate/reallocate diffe rent fields on the go, freeing memory or keeping it for another use - basically like pytorch - without having to reinit)

- multiple runtimes

- fields, data oriented class in aot

- some sort of thread lock (like cuda atomicCAS)

Cheers!
